main.tf

```
@@ -21,4 +30,11 @@ resource "aws_cloudwatch_event_target" "this" {
       input_template = input_transformer.value.input_template
     }
   }
+
+  dynamic "sqs_target" {
+    for_each = each.value.sqs_target != null ? [1] : []
+    content {
+      message_group_id = each.value.sqs_target.message_group_id
+    }
+  }
```

same here, use the dynamic block as the iteration object

```suggestion
  dynamic "sqs_target" {
    for_each = each.value.sqs_target != null ? [each.value.sqs_target] : []
    content {
      message_group_id = sqs_target.value.message_group_id
    }
  }
```